### ISSUE TITLE
issue installing foundation cli windows 10

### ISSUE THREAD

**Issue by geraldinethomas5** [zurb/foundation-sites#7506]
hello,
I'm trying to run $fundation new  and I get an error as follows:

Downloading the project template...
Done downloading!

Installing dependencies...

assert.js:86
  throw new assert.AssertionError({
        ^
AssertionError: null == true
    at TLSSocket.socketOnData (_http_client.js:308:3)
    at TLSSocket.emit (events.js:107:17)
    at TLSSocket.Readable.read (_stream_readable.js:373:10)
    at TLSSocket.socketCloseListener (_http_client.js:229:10)
    at TLSSocket.emit (events.js:129:20)
    at TCP.close (net.js:476:12)

what does this error mean? thank you!

**Comment by gakimball** [comment]
What settings did you use? Framework, project name, and template? Also, what shell do you use? The basic Windows Command Prompt or something else?

**Comment by geraldinethomas5** [comment]
Hi,
Im on windows 10 64 bit
npm v 2.5.1
node v 0.12.0
ran $foundation new
then chose foundation for sites
next entered name of my project
and next i chose zurb Template with the handlebars and scss and js
compilers.

I use both regular command prompt from my machine and the cmder program
that uses powershell.exe that I installed.
I received same error on both command lines.

(I had previously had zf5 generator with yeoman, but ran $gem uninstall
foundation and the new cli installation command after i made sure it was
wiped off my system)

Thanks for the help!!

**Comment by gakimball** [comment]
In your original comment, is that the full text of the error you got? The stack trace doesn't include any reference to the Foundation CLI itself, which makes it difficult to debug.

If there's a longer error message you can post, that would be helpful. If not, I have other ways we can debug the issue.

**Comment by geraldinethomas5** [comment]
yes that was all it got as far as 

installing dependencies...

then it I got that error.

**Comment by geraldinethomas5** [comment]
hi again, I went ahead and updated the node.js version on my machine to 5.2.0 and ran 

$foundation new

just to see if problems would be solved. and I got this error message.  the cli creates the project but can not download npm modules here is what happened:




C:\Users\geraldine>node -v
v5.2.0

C:\Users\geraldine>foundation new
? What are you building today? A website (Foundation for Sites)
? What's the project called? gdemosite
? Which template would you like to use? ZURB Template: includes Handlebars templa
tes and Sass/JS compilers

            .
           /|     ,
      , /|/  \/| /|       Thanks for using ZURB Foundation for Sites!
     /|/       |/ |       -------------------------------------------
 |___|            |___|   Let's set up a new project.
 \___|  ^^   ^^   |___/   It shouldn't take more than a minute.
     | -[O]--[O]- |
     |    ___,    |
     |    ...     |
      \__________/


Downloading the project template...
Done downloading!

Installing dependencies...


> phantomjs@1.9.19 install C:\Users\geraldine\gdemosite\node_modules\gulp-uncss\n
ode_modules\uncss\node_modules\phridge\node_modules\phantomjs
> node install.js

PhantomJS not found on PATH
Download already available at C:\Users\GERALD~1\AppData\Local\Temp\phantomjs\phan
tomjs-1.9.8-windows.zip
Verified checksum of previously downloaded file
Extracting zip contents
Removing C:\Users\geraldine\gdemosite\node_modules\gulp-uncss\node_modules\uncss\
node_modules\phridge\node_modules\phantomjs\lib\phantom
Copying extracted folder C:\Users\GERALD~1\AppData\Local\Temp\phantomjs\phantomjs
-1.9.8-windows.zip-extract-1450126289809\phantomjs-1.9.8-windows -> C:\Users\gera
ldine\gdemosite\node_modules\gulp-uncss\node_modules\uncss\node_modules\phridge\n
ode_modules\phantomjs\lib\phantom
Writing location.js file
Done. Phantomjs binary available at C:\Users\geraldine\gdemosite\node_modules\gul
p-uncss\node_modules\uncss\node_modules\phridge\node_modules\phantomjs\lib\phanto
m\phantomjs.exe

> utf-8-validate@1.2.1 install C:\Users\geraldine\gdemosite\node_modules\browser-
sync\node_modules\socket.io\node_modules\engine.io\node_modules\ws\node_modules\u
tf-8-validate
> node-gyp rebuild

\
C:\Users\geraldine\gdemosite\node_modules\browser-sync\node_modules\socket.io\nod
if not defined npm_config_node_gyp (node "C:\Users\geraldine\AppData\Roaming\npm\
node_modules\foundation-cli\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules
(node  rebuild ) e-gyp.js" rebuild ) / else
gyp ERR! configure error
gyp ERR! stack Error: Python executable "C:\Users\geraldine\AppData\Local\Program
s\Python\Python35\python.EXE" is v3.5.1, which is not supported by gyp.
gyp ERR! stack You can pass the --python switch to point to Python >= v2.5.0 & <
3.0.0.
gyp ERR! stack     at failPythonVersion (C:\Users\geraldine\AppData\Roaming\npm\n
ode_modules\foundation-cli\node_modules\npm\node_modules\node-gyp\lib\configure.j
s:406:14)
gyp ERR! stack     at C:\Users\geraldine\AppData\Roaming\npm\node_modules\foundat
ion-cli\node_modules\npm\node_modules\node-gyp\lib\configure.js:395:9
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:205:7)
gyp ERR! stack     at emitTwo (events.js:88:13)
ERR! stack     at ChildProcess.emit (events.js:173:7)
gyp ERR! stack     at maybeClose (internal/child_process.js:819:16)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process
.js:212:5)
gyp ERR! System Windows_NT 10.0.10586
gyp ERR! command\ "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\geraldine\\Ap
pData\\Roaming\\npm\\node_modules\\foundation-cli\\node_modules\\npm\\node_module
s\\node-gyp\\bin\\node-gyp.js" "rebuild"
 ERR! cwd C:\Users\geraldine\gdemosite\node_modules\browser-sync\node_modules\soc
ket.io\node_modules\engine.io\node_modules\ws\node_modules\utf-8-validate
 node -v v5.2.0
gyp ERR! node-gyp -v v3.2.1
gyp ERR! not ok

> bufferutil@1.2.1 install C:\Users\geraldine\gdemosite\node_modules\browser-sync
\node_modules\socket.io\node_modules\engine.io\node_modules\ws\node_modules\buffe
rutil
> node-gyp rebuild

-
C:\Users\geraldine\gdemosite\node_modules\browser-sync\node_modules\socket.io\nod
e_modules\engine.io\node_modules\ws\node_modules\bufferutil>if not defined npm_co
nfig_node_gyp (node "C:\Users\geraldine\AppData\Roaming\npm\node_modules\foundati
on-cli\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gy
 else (node  rebuild )
gyp ERR! stack You can pass the --python switch to point to Python >= v2.5.0 & <
3.0.0.R! stack Error: Python executable "C:\Users\geraldine\AppData\Local\Program
gyp ERR! stack     at failPythonVersion (C:\Users\geraldine\AppData\Roaming\npm\n
ode_modules\foundation-cli\node_modules\npm\node_modules\node-gyp\lib\configure.j
s:406:14)
stack     at C:\Users\geraldine\AppData\Roaming\npm\node_modules\foundation-cli\n
ode_modules\npm\node_modules\node-gyp\lib\configure.js:395:9
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:205:7)
gyp ERR! stack     at emitTwo (events.js:88:13)
gyp ERR! stack     at ChildProcess.emit (events.js:173:7)
gyp ERR! stack     at maybeClose (internal/child_process.js:819:16)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process
.js:212:5)
System Windows_NT 10.0.10586
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\geraldine\\App
Data\\Roaming\\npm\\node_modules\\foundation-cli\\node_modules\\npm\\node_modules
\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\geraldine\gdemosite\node_modules\browser-sync\node_modules\
socket.io\node_modules\engine.io\node_modules\ws\node_modules\bufferutil
gyp ERR! node -v v5.2.0
gyp ERR! node-gyp -v v3.2.1
gyp ERR! not ok

> gifsicle@3.0.3 postinstall C:\Users\geraldine\gdemosite\node_modules\gulp-image
min\node_modules\imagemin\node_modules\imagemin-gifsicle\node_modules\gifsicle
> node lib/install.js

  √ gifsicle pre-build test passed successfully

> optipng-bin@3.0.4 postinstall C:\Users\geraldine\gdemosite\node_modules\gulp-im
agemin\node_modules\imagemin\node_modules\imagemin-optipng\node_modules\optipng-b
in
> node lib/install.js

  √ optipng pre-build test passed successfully

> jpegtran-bin@3.0.6 postinstall C:\Users\geraldine\gdemosite\node_modules\gulp-i
magemin\node_modules\imagemin\node_modules\imagemin-jpegtran\node_modules\jpegtra
n-bin
> node lib/install.js

  √ jpegtran pre-build test passed successfully

> bufferutil@1.2.1 install C:\Users\geraldine\gdemosite\node_modules\browser-sync
\node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-clie
nt\node_modules\ws\node_modules\bufferutil
> node-gyp rebuild


C:\Users\geraldine\gdemosite\node_modules\browser-sync\node_modules\socket.io\nod
e_modules\socket.io-client\node_modules\engine.io-client\node_modules\ws\node_mod
ules\bufferutil>if not defined npm_config_node_gyp (node "C:\Users\geraldine\AppD
ata\Roaming\npm\node_modules\foundation-cli\node_modules\npm\bin\node-gyp-bin\\..
\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node  rebuild )
gyp ERR! configure error
gyp ERR! stack Error: Python executable "C:\Users\geraldine\AppData\Local\Program
s\Python\Python35\python.EXE" is v3.5.1, which is not supported by gyp.
gyp ERR! stack You can pass the --python switch to point to Python >= v2.5.0 & <
3.0.0.
gyp ERR! stack     at failPythonVersion (C:\Users\geraldine\AppData\Roaming\npm\n
ode_modules\foundation-cli\node_modules\npm\node_modules\node-gyp\lib\configure.j
s:406:14)
gyp ERR! stack     at C:\Users\geraldine\AppData\Roaming\npm\node_modules\foundat
ion-cli\node_modules\npm\node_modules\node-gyp\lib\configure.js:395:9
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:205:7)
gyp ERR! stack     at emitTwo (events.js:88:13)
gyp ERR! stack     at ChildProcess.emit (events.js:173:7)
gyp ERR! stack     at maybeClose (internal/child_process.js:819:16)
gyp ERR! stack     at Socket.<anonymous> (internal/child_process.js:320:11)
gyp ERR! stack     at emitOne (events.js:78:13)
gyp ERR! stack     at Socket.emit (events.js:170:7)
gyp ERR! stack     at Pipe._onclose (net.js:470:12)
gyp ERR! System Windows_NT 10.0.10586
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\geraldine\\App
Data\\Roaming\\npm\\node_modules\\foundation-cli\\node_modules\\npm\\node_modules
\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\geraldine\gdemosite\node_modules\browser-sync\node_modules\
socket.io\node_modules\socket.io-client\node_modules\engine.io-client\node_module
s\ws\node_modules\bufferutil
gyp ERR! node -v v5.2.0
gyp ERR! node-gyp -v v3.2.1
gyp ERR! not ok

> utf-8-validate@1.2.1 install C:\Users\geraldine\gdemosite\node_modules\browser-
sync\node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-
client\node_modules\ws\node_modules\utf-8-validate
> node-gyp rebuild


C:\Users\geraldine\gdemosite\node_modules\browser-sync\node_modules\socket.io\nod
e_modules\socket.io-client\node_modules\engine.io-client\node_modules\ws\node_mod
ules\utf-8-validate>if not defined npm_config_node_gyp (node "C:\Users\geraldine\
AppData\Roaming\npm\node_modules\foundation-cli\node_modules\npm\bin\node-gyp-bin
\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node  rebuild )
gyp ERR! configure error
gyp ERR! stack Error: Python executable "C:\Users\geraldine\AppData\Local\Program
s\Python\Python35\python.EXE" is v3.5.1, which is not supported by gyp.
gyp ERR! stack You can pass the --python switch to point to Python >= v2.5.0 & <
3.0.0.
gyp ERR! stack     at failPythonVersion (C:\Users\geraldine\AppData\Roaming\npm\n
ode_modules\foundation-cli\node_modules\npm\node_modules\node-gyp\lib\configure.j
s:406:14)
gyp ERR! stack     at C:\Users\geraldine\AppData\Roaming\npm\node_modules\foundat
ion-cli\node_modules\npm\node_modules\node-gyp\lib\configure.js:395:9
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:205:7)
gyp ERR! stack     at emitTwo (events.js:88:13)
gyp ERR! stack     at ChildProcess.emit (events.js:173:7)
gyp ERR! stack     at maybeClose (internal/child_process.js:819:16)
gyp ERR! stack     at Socket.<anonymous> (internal/child_process.js:320:11)
gyp ERR! stack     at emitOne (events.js:78:13)
gyp ERR! stack     at Socket.emit (events.js:170:7)
gyp ERR! stack     at Pipe._onclose (net.js:470:12)
gyp ERR! System Windows_NT 10.0.10586
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\geraldine\\App
Data\\Roaming\\npm\\node_modules\\foundation-cli\\node_modules\\npm\\node_modules
\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\geraldine\gdemosite\node_modules\browser-sync\node_modules\
socket.io\node_modules\socket.io-client\node_modules\engine.io-client\node_module
s\ws\node_modules\utf-8-validate
gyp ERR! node -v v5.2.0
gyp ERR! node-gyp -v v3.2.1
gyp ERR! not ok

> spawn-sync@1.0.13 postinstall C:\Users\geraldine\gdemosite\node_modules\gulp-sa
ss\node_modules\node-sass\node_modules\cross-spawn\node_modules\spawn-sync
> node postinstall


> node-sass@3.4.2 install C:\Users\geraldine\gdemosite\node_modules\gulp-sass\nod
e_modules\node-sass
> node scripts/install.js

Binary downloaded and installed at C:\Users\geraldine\gdemosite\node_modules\gulp
-sass\node_modules\node-sass\vendor\win32-x64-47\binding.node

> node-sass@3.4.2 postinstall C:\Users\geraldine\gdemosite\node_modules\gulp-sass
\node_modules\node-sass
> node scripts/build.js

` C:\Users\geraldine\gdemosite\node_modules\gulp-sass\node_modules\node-sass\vend
or\win32-x64-47\binding.node ` exists.
 testing binary.
Binary is fine; exiting.

There were some problems during the installation.

 ✓ New project folder created.
 ✗ Node modules not installed. Try running npm install manually.
 ✓ Bower components installed.

Once you've resolved the above issues, run foundation watch while inside the gdem
osite folder.

I went ahead and cd into project


C:\Users\geraldine\gdemosite>foundation watch

> foundation-zurb-template@1.0.0 start C:\Users\geraldine\gdemosite
> gulp

module.js:329
    throw err;
    ^

Error: Cannot find module 'core-util-is'
    at Function.Module._resolveFilename (module.js:327:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:355:17)
    at require (internal/module.js:13:17)
    at Object.<anonymous> (C:\Users\geraldine\gdemosite\node_modules\gulp\node_modules\vinyl-fs\node_modules\through2\node_m
odules\readable-stream\lib\_stream_duplex.js:39:12)
    at Module._compile (module.js:399:26)
    at Object.Module._extensions..js (module.js:406:10)
    at Module.load (module.js:345:32)
    at Function.Module._load (module.js:302:12)
    at Module.require (module.js:355:17)
 Error: foundation-zurb-template@1.0.0 start: `gulp`
Exit status 1
    at EventEmitter.<anonymous> (C:\Users\geraldine\AppData\Roaming\npm\node_modules\foundation-cli\node_modules\npm\lib\uti
ls\lifecycle.js:214:16)
    at emitTwo (events.js:88:13)
    at EventEmitter.emit (events.js:173:7)
    at ChildProcess.<anonymous> (C:\Users\geraldine\AppData\Roaming\npm\node_modules\foundation-cli\node_modules\npm\lib\uti
ls\spawn.js:24:14)
    at emitTwo (events.js:88:13)
    at ChildProcess.emit (events.js:173:7)
    at maybeClose (internal/child_process.js:819:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:212:5)

and also $npm start


C:\Users\geraldine\gdemosite>npm start

> foundation-zurb-template@1.0.0 start C:\Users\geraldine\gdemosite
> gulp

module.js:329
    throw err;
    ^

Error: Cannot find module 'core-util-is'
    at Function.Module._resolveFilename (module.js:327:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:355:17)
    at require (internal/module.js:13:17)
    at Object.<anonymous> (C:\Users\geraldine\gdemosite\node_modules\gulp\node_modules\vinyl-fs\node_modules\through2\node_
odules\readable-stream\lib\_stream_duplex.js:39:12)
    at Module._compile (module.js:399:26)
    at Object.Module._extensions..js (module.js:406:10)
    at Module.load (module.js:345:32)
    at Function.Module._load (module.js:302:12)
    at Module.require (module.js:355:17)
    at require (internal/module.js:13:17)

npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\geraldine\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\n
m-cli.js" "start"
npm ERR! node v5.2.0
npm ERR! npm  v3.3.9
npm ERR! code ELIFECYCLE
npm ERR! foundation-zurb-template@1.0.0 start: `gulp`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the foundation-zurb-template@1.0.0 start script 'gulp'.
npm ERR! This is most likely a problem with the foundation-zurb-template package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     gulp
npm ERR! You can get their info via:
npm ERR!     npm owner ls foundation-zurb-template
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\geraldine\gdemosite\npm-debug.log

Thanks again!

**Comment by gakimball** [comment]
At some point in the install process, a module failed to install properly, although it's hard to say which.

Try just running `npm install` inside your project folder to see if it missed anything.

I also see a Gulp-specific dependency error, so after that you can also try `npm install gulp`.

**Comment by geraldinethomas5** [comment]
Hi! my project works perfectly now! All along I have been trying to choose zurb Template right off the bat, and never tested with just Basic Template. I did I chose that and ran it and it worked perfect. then I ran 
$foundation new again and chose the Zurb Template and this time it worked everything is installed and it launches the project in the browser. I am not sure what changed except that I ran Basic first and now the other works. 
![screenshot 2015-12-16 20 59 56](https://cloud.githubusercontent.com/assets/12564675/11859457/1441a888-a439-11e5-8990-c52dc1f490f0.png)
 Thanks so much for your time helping me troubleshoot!
Best Regards!
